### PR TITLE
Fix bug where Que::JobTimeoutError is not retried

### PR DESCRIPTION
**Pull request by lawrencejones**
An accidental .dup prevented Que::JobTimeoutError from ever being added
to the list of retryable exceptions. In addition, if
retryable_exceptions were never specified then we wouldn't retry the
timeout, which we want to do in all cases.

### CODE REVIEW

File: lib/que/failure/strategies/variable_retry.rb [hunk 1]
```diff
@@ -108,6 +105,12 @@ def handle_job_failure(error, job)
         private
 
         def retryable_exception?(error)
+          # This is a special class of error returned by Que if the job failed respond
+          # promptly to a stop signal. We should assume these exceptions are always
+          # retryable, to prevent a scheduler from terminating Que midway through the work
+          # loop.
```
Comment (hmac) [inline lib/que/failure/strategies/variable_retry.rb]: I don't think this reason is accurate. It's not about Que being terminated during the work loop - it's just to ensure that timeouts don't interfere with retry strategies.

Comment (lawrencejones) [reply lib/que/failure/strategies/variable_retry.rb]: I should've been clearer: this comment should have a "which interferes with retry strategies" suffixed to it. Will ad..

Comment (hmac) [reply lib/que/failure/strategies/variable_retry.rb]: I still don't think that's clear. It's the exception itself that ensures Que cleans up correctly. This change just makes it play nicely with retry strategies. I would have said something like

```
This is a special class of exception raised by Que if the worker failed to respond promptly to a stop signal. These aren't directly caused by the job itself and so we always treat them as retryable.
This preserves the existing behaviour of `VariableRetry` in response to process termination.
```

Comment (lawrencejones) [reply lib/que/failure/strategies/variable_retry.rb]: That works, will use that. Thanks!

**Comment by lawrencejones** [comment]
Take 2

**Pull request merged**